### DISCUSSION TITLE
Livewire: undocumented stuff

### DISCUSSION THREAD

**Discussion by tpaksu** [livewire/livewire#1212] [4. General]
Hello, 

I was playing with `wire:click` and `wire:loading` and I realized there are missing stuff on that feature.

1. You can't set an interest area to the loader, for example, you have a toggle button grid and on every cell, you have a toggle button. and you only want to replace the clicked toggle button with the loading indicator, but at the current state of the plugin, every toggle button is replaced with the loading indicator. 

2. There seems an undocumented vital point where `wire:loading` attributes **should be written BEFORE** `wire:click` to get to work. Otherwise, it simply doesn't work. 

3. Also, removing and adding multiple attributes/classes with `wire:loading.class.remove` isn't documented. But writing space separated classes work. Might be good to be documented to prevent confusion. 

4. If I have passed variables through `wire:click`, like 

```php
<i
    wire:loading.class.remove="fa-check cursor-pointer text-green-600"
    wire:loading.class="cursor-wait fa-hourglass text-gray-300"
    wire:click="togglePermission({{$role->id}},{{$permission->id}})"
    class="fa fa-check text-lg font-bold text-green-600 cursor-pointer"></i>
```

is there a chance that I directly get models on the `togglePermission` method? Like

```php
    public function togglePermission(Role $role, Permission $permission) { ...
``` 

I'll try to write here if I'll bump into more situations that I feel might be good to write down somewhere.

**Comment by nuernbergerA** [comment]
Hey @tpaksu 

could you open an issue for your 2nd point with an easy code example?

Cheers Adrian

**Reply by tpaksu** [reply]
hi @nuernbergerA , 

I created a repo which you can directly clone and test. 

https://github.com/tpaksu/livewire-test

I tried and it worked at both positions, but in my case it was inside an **if** block. 

In the demo, I created the example from the documentation itself and modified it, the up arrow icon should replace with a grey hourglass and it does, but the down arrow in the else case it doesn't. If you remove the if clause, both works. 

It wasn't related with the order it seems.

**Reply by tpaksu** [reply]
Component:
----
https://github.com/tpaksu/livewire-test/blob/584c0086a65444eac4d5dedd3b2a690da51241ee/app/Http/Livewire/Counter.php#L1-L27

View:
---
https://github.com/tpaksu/livewire-test/blob/584c0086a65444eac4d5dedd3b2a690da51241ee/resources/views/livewire/counter.blade.php#L1-L18

**Comment by nuernbergerA** [comment]
It should work with a key

```blade
<div style="text-align: center">
    <h1>{{ $count }}</h1>

    @if($count % 2 === 0)
    <i
        wire:key="up"
        wire:loading.class.remove="fa-chevron-up cursor-pointer text-green-600"
        wire:loading.class="cursor-wait fa-hourglass text-gray-300"
        wire:click="increment"
        class="fa fa-chevron-up text-lg font-bold text-green-600 cursor-pointer"></i>
    @else
    <i
        wire:key="down"
        wire:click="decrement"
        wire:loading.class.remove="fa-chevron-down cursor-pointer text-green-600"
        wire:loading.class="cursor-wait fa-hourglass text-gray-300"
        class="fa fa-chevron-down text-lg font-bold text-green-600 cursor-pointer"></i>
    @endif
</div>
```

**Reply by tpaksu** [reply]
Yes it works, <strike>but still it is undocumented. Is it because the if clause? When the ajax replaces the element, is an event handler not registered which listens for the loading event? I should read the source code I guess. But I see "adding a key works" everywhere. Is there something related with it in the docs?</strike> Yes, it's documented in the [troubleshooting](https://laravel-livewire.com/docs/troubleshooting) page.

**Reply by tpaksu** [reply]
@nuernbergerA you marked this as answered, but this wasn't the question.
